server/src/main/java/org/elasticsearch/common/io/stream/StreamInput.java

```diff
@@ -738,8 +738,8 @@ public Object readGenericValue() throws IOException {
             case 6 -> readByteArray();
             case 7 -> readArrayList();
             case 8 -> readArray();
-            case 9 -> readLinkedHashMap();
-            case 10 -> readHashMap();
+            case 9 -> version.onOrAfter(Version.V_8_4_0) ? readObjectKeyLinkedHashMap() : readLinkedHashMap();
```

Comment:
Why are new methods necessary? I think even the existing linkedhashmap methods can be replaced:

generic linked map:
```
readOrderedMap(StreamInput::readGenericValue, StreamInput::readGenericValue);
```

string -> object linked map
```
readOrderedMap(StreamInput::readString, StreamInput::readGenericValue);
```

And for non linked, the same but with readMap

Reply:
Ah! I just left a similar comment. FWIW if there are lots of usages of `readLinkedHashMap` then I might bump removing the method to a follow up. But, yeah, this'd be fine:
```
readOrderedMap(StreamInput::readGenericValue, StreamInput::readGenericValue);
```